src/mds/Server.cc

```diff
@@ -5932,25 +6113,29 @@ void Server::handle_client_removexattr(MDRequestRef& mdr)
   if (!mds->locker->acquire_locks(mdr, lov))
     return;
 
+
+  auto handler = Server::get_xattr_or_default_handler(name);
+  bufferlist bl;
+  XattrOp xattr_op(CEPH_MDS_OP_RMXATTR, name, bl, 0);
+
   const auto& pxattrs = cur->get_projected_xattrs();
-  if (pxattrs && pxattrs->count(mempool::mds_co::string(name)) == 0) {
-    dout(10) << "removexattr '" << name << "' and ENODATA on " << *cur << dendl;
-    respond_to_request(mdr, -ENODATA);
+  int r = (this->*(handler->validate))(cur, pxattrs, &xattr_op);
```

Comment:
ditto, this syntax is also unknown to me.

Reply:
https://isocpp.org/wiki/faq/pointers-to-members#dotstar-vs-arrowstar

(habit of using shorthand -- maybe just switch to using std::invoke)

Reply:
Yes, please do that instead.